modules/fedimint-ln-tests/tests/tests.rs

```diff
@@ -452,6 +454,100 @@ async fn makes_internal_payments_within_federation() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn can_receive_for_other_user() -> anyhow::Result<()> {
+    let fixtures = fixtures();
+    let fed = fixtures.new_fed().await;
+    let (client1, client2) = fed.two_clients().await;
+    let client2_dummy_module = client2.get_first_module::<DummyClientModule>();
+
+    // generate a new keypair
+    let keypair = KeyPair::new_global(&mut OsRng);
+
+    // Print money for client2
+    let (op, outpoint) = client2_dummy_module.print_money(sats(1000)).await?;
+    client2.await_primary_module_output(op, outpoint).await?;
+
+    // TEST internal payment when there are no gateways registered
+    let (recv_op, invoice, _) = client1
+        .get_first_module::<LightningClientModule>()
+        .create_bolt11_invoice_for_user(
+            sats(250),
+            "for user".to_string(),
+            None,
+            keypair.public_key(),
+            (),
+        )
+        .await?;
+    let mut sub1 = client1
+        .get_first_module::<LightningClientModule>()
+        .subscribe_ln_receive(recv_op)
+        .await?
+        .into_stream();
+    assert_eq!(sub1.ok().await?, LnReceiveState::Created);
+    assert_matches!(sub1.ok().await?, LnReceiveState::WaitingForPayment { .. });
+
+    let OutgoingLightningPayment {
+        payment_type,
+        contract_id: _,
+        fee: _,
+    } = pay_invoice(&client2, invoice.clone()).await?;
+    match payment_type {
+        PayType::Internal(op_id) => {
+            let mut sub2 = client2
+                .get_first_module::<LightningClientModule>()
+                .subscribe_internal_pay(op_id)
+                .await?
+                .into_stream();
+            assert_eq!(sub2.ok().await?, InternalPayState::Funding);
+            assert_matches!(sub2.ok().await?, InternalPayState::Preimage { .. });
+            // goes from preimage to immediate claim because it is for another user
+            assert_eq!(sub1.ok().await?, LnReceiveState::Claimed);
+        }
+        _ => panic!("Expected internal payment!"),
+    }
+
+    // TEST internal payment when there is a registered gateway
+    gateway(&fixtures, &fed).await;
+
+    let (op, invoice, _) = client1
+        .get_first_module::<LightningClientModule>()
+        .create_bolt11_invoice(sats(250), "with-gateway-hint".to_string(), None, ())
```

Comment:
Should this be `create_bolt11_invoice_for_user`?

Reply:
woops, fixed. Didn't realize it did the same thing twice, with and without the gateway